libdevcore/LruCache.h

```diff
@@ -0,0 +1,128 @@
+// Aleth: Ethereum C++ client, tools and libraries.
+// Copyright 2019 Aleth Authors.
+// Licensed under the GNU General Public License, Version 3.
+
+#pragma once
+
+#include <list>
+#include <unordered_map>
+
+namespace dev
+{
+template <class Key, class Value>
+class LruCache
+{
+    typedef Key key_type;
+    typedef Value value_type;
+    typedef std::list<std::pair<key_type, value_type>> list_type;
+    typedef std::unordered_map<key_type, typename list_type::const_iterator> map_type;
+
+public:
+    explicit LruCache(size_t _capacity) : m_capacity(_capacity) {}
+
+    LruCache(LruCache<key_type, value_type> const& _l)
```

Comment:
I think default-generated copy and move constructors and assignment operators should be enough.